subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/UnitOfWorkProperties.java

```diff
@@ -37,14 +32,14 @@
  *
  * Once created, the view is immutable and registering additional or changing existing task properties will not be detected.
  *
- * Created by {@link DefaultTaskProperties#resolve(PropertyWalker, PathToFileResolver, TaskInternal)}.
+ * Created by {@link DefaultUnitOfWorkProperties#resolve(PropertyWalker, org.gradle.api.internal.file.FileResolver, TaskInternal)}.
  */
 @NonNullApi
-public interface TaskProperties {
+public interface UnitOfWorkProperties {
```

Comment:
These are all bound to tasks still. I don't see a reason to rename the type to `UnitOfWorkProperties`. Instead, when we move this functionality to the execution engine, this type should be gone entirely (or at least reduced to contain information only about the output properties).

Reply:
That is fine with me, I don't have to do the rename now. The main reason for doing the rename was that the whole type is not bound to tasks anymore.

Reply:
OK, let's revert the rename then. We should introduce only the necessary stuff to the `:execution` model, so moving things over there one-by-one as needed seems like a better approach.

Reply:
I reverted the name. I left the class in `task.properties` for now.